pacstall

```diff
@@ -530,8 +531,37 @@ for i in "${@}"; do
     unset j
 done
 
-# Set the new list of arguments
-set -- "${argument_list[@]}"
+# Remove duplicates
+declare -A arg_map=(
+    ["--install"]="-I"     ["--search"]="-S"   ["--remove"]="-R"          ["--download"]="-D"
+    ["--add-repo"]="-A"    ["--update"]="-U"   ["--list"]="-L"            ["--upgrade"]="-Up"
+    ["--query-info"]="-Qi" ["--tree"]="-T"     ["--version"]="-V"         ["--help"]="-h"
+    ["--build"]="-B"       ["--keep"]="-K"     ["--disable-prompts"]="-P" ["--nocheck"]="-Nc"
+)
+declare -A hashed_arguments
+unique_argument_list=()
+for arg in "${argument_list[@]}"; do
+    mapped="${arg_map[$arg]:=$arg}"
+    if ! [[ -v "hashed_arguments[$mapped]" ]]; then
+        unique_argument_list+=($arg)
+        hashed_arguments[$mapped]=0
+    fi
+done
+unset hashed_arguments
+
+# Check if only one command flag is being used
+short_commands=("-I" "-S" "-R" "-D" "-A" "-U" "-L" "-Up" "-Qi" "-T" "-V" "-h")
+long_commands=("--install" "--search" "--remove" "--download" "--add-repo" "--update" "--list" "--upgrade" "--query-info" "--tree" "--version" "--help")
+commands=("${short_commands[@]}" "${long_commands[@]}")
+matches=($(comm -12 <(sort <(printf "%s\n" "${unique_argument_list[@]}")) <(sort <(printf "%s\n"  "${commands[@]}"))))
```

Comment:
can you explain what im looking at here please

Reply:
Compare the list of deduplicated args with the full list of command flags.

If there is more than one match, that means you are trying to run multiple commands at the same time.